package.json

```diff
@@ -0,0 +1,29 @@
+{
+  "name": "webdesign-project",
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
+  "dependencies": {
+    "@types/react-router-dom": "^5.3.3",
+    "axios": "^1.7.7",
+    "cors": "^2.8.5",
+    "dotenv": "16.4.5",
+    "express": "^4.21.1",
+    "mongoose": "^8.7.1",
```

Comment:
should some modules, such as dotenv, express, and mongoose, be installed to ./backend/package.json file?

Reply:
If they have been install at ./backend/package.json file, why keep these modules here?

Reply:
Yes I need them for the backend in the  different .ts files.
The frontend and backend are run through concurrently.

When running npm run dev in the backend directory it starts both the front end "npm start --prefix ../frontend/client"
and backend "ts-node server.ts"

All of the dependencies in the package.json file are being used or will be used at some point.

Reply:
> Yes I need them for the backend in the different .ts files. The frontend and backend are run through concurrently.
> 
> When running npm run dev in the backend directory it starts both the front end "npm start --prefix ../frontend/client" and backend "ts-node server.ts"
> 
> All of the dependencies in the package.json file are being used or will be used at some point.

Right now you have three package.json files: one under /backend/, second one under /frontend, and then you created the third one under root. I cannot see why package.json under root is needed. The /backend/package.json file is used for installing modules for the nodejs/express server, whereas the /frontend/package.json file is used to install modules needed for the frontend.